lib/Image/PNG/Portable.rakumod

```diff
@@ -78,13 +79,34 @@ method write(Str:D $file) {
 
     write-chunk $fh, 'IDAT', compress $!data;
 
+    if %!text-metadata.defined {
+        for %!text-metadata.kv -> $key, $value {
+            my @data = flat $key.encode('latin-1').list, 0, $value.encode('latin-1').list;
+            write-chunk $fh, 'tEXt', @data;
+        }
+    }
+
     write-chunk $fh, 'IEND';
 
     $fh.close;
 
     True
 }
 
+multi method set-text-meta(Str:D $key, Str:D $value) {
+    fail "Invalid tEXt keyword (must be 1–79 Latin-1 chars)" 
```

Comment:
`if !$key || $key.contains(/<:!Script<Latin>>/) || $key.chars > 79`

If there is no key or contains any non-Latin characters or is longer than 79 chars